### PR TITLE
flux-jobs/flux-pgrep: rename --suppress-header to --no-header

### PR DESCRIPTION
**Pull request by chu11**
Per #4899, make the `--suppress-header` / `--no-header` consistent between tools.  Chose to common-ify on `--no-header`.  

Note that I legacy support `--suppress-header` in `flux-jobs` but not in `flux-pgrep`, b/c the latter has not officially been released yet, so I assumed no users of it yet.

### CODE REVIEW

File: src/cmd/flux-jobs.py
Comment (grondo) [inline src/cmd/flux-jobs.py:266]: you can use the [dest](https://docs.python.org/3/library/argparse.html#dest) option to `add_argument` to store true in `no_header` instead of `suppress_header` so that this just becomes a hidden alternate for `--no-header`, e.g.

```python
parser.add_argument(
    "--suppress-header",
    action="store_true",
    dest="no_header",
    help=argparse.SUPPRESS
)
```

Then remove references to `args.suppress_header` everwhere.

**Review by grondo — COMMENTED** [review]
Looks good. One suggested change inline.

**Comment by chu11** [comment]
thanks!  re-pushed with the tiny cleanup and setting MWP

**Review by grondo — APPROVED** [review]
LGTM! (you can put merge-when-passing on a PR without approving review since the approval is one of the things mergify checks before merging)

**Pull request merged**
